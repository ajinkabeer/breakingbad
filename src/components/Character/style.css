.card {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
}

.card img {
  width: 100%;
}

.card-container {
  padding: 6px 16px;
  background-color: #fff;
}

.card:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
  cursor: pointer;
}

.card-title-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}
