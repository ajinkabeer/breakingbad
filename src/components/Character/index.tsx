import './style.css';

interface Props {
  id: number;
  name: string;
  img: string;
  portrayed: string;
  onClick: () => void;
}

const Character = ({ id, name, img, portrayed, onClick }: Props) => (
  <div className="card">
    <a
      onClick={onClick}
      tabIndex={id}
      onKeyPress={onClick}
      role="button"
      data-testid="click-character"
    >
      <img src={img} height={400} alt={name} />
      <div className="card-container">
        <div className="card-title-container">
          <h3>{name}</h3>
          <p>{portrayed}</p>
        </div>
      </div>
    </a>
  </div>
);

export default Character;
