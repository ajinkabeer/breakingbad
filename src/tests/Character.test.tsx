import { fireEvent, render } from '@testing-library/react';
import Character from '../components/Character';

const props = {
  id: 1,
  name: 'Name',
  img: 'test.png',
  portrayed: 'test',
  onClick: jest.fn(),
};
const { id, name, img, portrayed, onClick } = props;

describe('Character Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Character
        key={id}
        id={id}
        name={name}
        img={img}
        portrayed={portrayed}
        onClick={onClick}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('Test whether component can be clicked', () => {
    const { getByTestId } = render(
      <Character
        key={id}
        id={id}
        name={name}
        img={img}
        portrayed={portrayed}
        onClick={onClick}
      />
    );

    fireEvent.click(getByTestId('click-character'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
