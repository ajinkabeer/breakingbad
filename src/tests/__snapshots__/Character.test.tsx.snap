// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <a
      data-testid="click-character"
      role="button"
      tabindex="1"
    >
      <img
        alt="Name"
        height="400"
        src="test.png"
      />
      <div
        class="card-container"
      >
        <div
          class="card-title-container"
        >
          <h3>
            Name
          </h3>
          <p>
            test
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
